posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,797 @@
+
+**Regarding prioritization and focus:** Other infrastructure projects have drawn much of Mark’s attention (crater, triagebot) in the last few months.
+We had long-standing debt on Crater and triagebot that needed to be addressed, and performance work was (and is) seeing more investment from other folks so was in less need of direct attention.
+
+@rylev believes that compiler performance remains, in their opinion, the largest and most persistent problem for Rust.
+
+### Compiler Backend Aspirations (🛠️, 👩‍💻)
+
+#### Ease writing new backends via intrinsic MIR fallbacks
+
+<!-- https://hackmd.io/@rust-compiler-team/HylMg2m_9 -->
+
+**Progress:** The Rust project has made progress on this ambition since the start of the year
+
+**Goals:** We had no goals for this ambition planned for this year, but we made ad-hoc progress on the problem itself.
+We think a solution will be available in the next two years, but not sooner than six months.
+
+**How it started:** At start of 2022, we had a solution or goal in mind, but did not have a plan for how to achieve it.
+
+**How it's going:** We would like help deciding what to do next. Also, it would be wonderful to get
+mentoring instructions (even just a sketch of them) so that non-experts could
+help make progress on this
+
+**Details:** While we’ve technically made non-zero progress, it’s minimal and unstructured.
+
+**Regarding new contributors:** There are no contributors who have or had committed to making progress on this. A compiler team member did make a step forward on this at one point, but they are unsurprisingly busy with other things.
+
+Mentoring instructions, or even steps with minimal details, might help non-experts pick up work in the area.
+
+
+#### Cranelift
+
+<!-- https://hackmd.io/@rust-compiler-team/BJLre2Xu9 -->
+
+**Progress:** The Rust project has made progress on this ambition since the start of the year.
+
+**Goals:** We had no goals for this ambition planned for this year, but we made ad-hoc progress on the problem itself.
+With regards to the overall problem: In terms of getting cg_clif distributed with rustup, @bjorn3 thinks 6 months is feasible. In terms of implementing all missing rust features they think it will take longer.
+
+**How it started:** at start of 2022, we had a plan, some of which had been completed, but more work remained to be done.
+
+**How it's going:** we have a high-level end-to-end plan to solve the problem, and we have completed some of our milestones.
+
+**Regarding new contributors:** There have been a couple minor PR’s. There has also been someone potentially interested in contributing more, but given that most of the low hanging fruit has already been done by @bjorn3, they were not able to help the newcomer find an area to contribute to.
+
+**Regarding prioritization and focus:** @bjorn3 has kept their focus on working on `cg_clif` and rustc PR’s that are part of the time related to `cg_clif`.
```

Comment:
This sentence is gramatically incorrect.

Reply:
You're right.

```suggestion
**Regarding prioritization and focus:** @bjorn3 has kept their focus on working on `cg_clif` and rustc PR’s that are related to `cg_clif`.
```

Is this better?

Reply:
Yes